UV excess in low mass stars

cpsanchez opened the pull request:
New branch for "UV excess in low mass stars"

cpsanchez committed:
Create trial.txt

cpsanchez committed:
Add files via upload

cpsanchez committed:
Delete trial.txt

cpsanchez committed:
Add files via upload

Updates to old notebook:

- Changed SMC target to new one
- Extended wavelength range studied to better see the differences between SMC and LMC extinction curves
- Created exercise for the read to see (with solutions)
- Added citations/additional resources

cpsanchez committed:
Add files via upload

Updates to old notebook:

- Added new citations
- Included SIMBAD query using astroquery
- Started included parametrization of curves (not finished, LMC fit not correct yet)

cpsanchez committed:
Add files via upload

Updates to old notebook:

- Added new citations
- Fixed parametrization of curves 
          * Note: SMC target chosen has sparse data, which makes data-fitting not optimal, maybe example target should be changed
- Fixed NaN values in data
- Added documentation to functions
- Small changes to overall structure and outputs format

cpsanchez committed:
Add files via upload

Initial draft for tutorial UV_excess_in_low_mass_stars. 

So far:
- Initial structure of notebook
- Tentative learning goals
- Brief intro in light curves and csv files
- Example of light curve with gPhoton (standard example to learn to use gPhoton, not related to the goal of notebook)
- Additional resources + citations

cpsanchez committed:
Merge pull request #1 from cpsanchez/UV_excess_in_low_mass_stars

Add files via upload

cpsanchez committed:
Delete UV_excess_in_low_mass_stars.ipynb

cpsanchez committed:
Create new repo

cpsanchez committed:
Add files via upload

Initial draft for UV_excess_for_low_mass_stars

So far:
- Initial structure of notebook
- Tentative learning goals 
- Short introduction on light curves and csv files
- Create and plot light curve using gPhoton (standard example to learn how to use gPhoton, not related with goal of tutorial)
- Additional resources + citations + imports

cpsanchez committed:
Delete trial.txt

cpsanchez committed:
Delete exploring_UV_extinction_curves directory

kelle commented:
This looks like a good start but I'm not sure I'm following the story. The learning goals are about UV excess but than the content is about light curves.  Let us know if you need help or when it's ready for review.

cpsanchez committed:
Add files via upload

Updates:

- GALEX introduction
- gFind to check if target is in database
- gAperture to download light curve

cpsanchez commented:
Indeed I still want some guidance on how exactly we want to characterize and study the UV excess, Scott has told me that David Rodriguez may be the person to ask about this, but he is on holidays until the 14th, so for now I am looking online for different articles that may be relevant and completing the parts about accessing GALEX using gPhoton!

cpsanchez committed:
New tutorial proposal

Includes:

- Data access to GUVcat through author's webpage
- Data downloading from URL
- Data accessing with pandas for .csv files

cpsanchez committed:
Add files via upload

Updates from last version: 

- Data reduction included from original catalog
        Criteria:
        * Source has both NUV and FUV magnitudes
        * Only one source within 2.5"
        * Not an extended object
- Color-color diagram

cpsanchez committed:
Add files via upload

Updates: 

- Target: TWA
- Search by coordinates in GUVCat: NUV magnitude
- Search by ID in Simbad: V, K magnitude
- Match in case of missing value
- Color-color diagram

Problem: GUVCat from author's webpage is divided each 5 degrees in galactic latitude, and TWA is dispersed over wider ranges. Additionally, each 5 degrees catalog takes up to an hour to download from URL.

cpsanchez committed:
Add files via upload

New tutorial structure based on David Rodriguez's guidance:

- Access GUVCat via the Virtual Observatory
- Look up 1 source and explore data provided by VO
- Look up all TWA and store NUV magnitude
- Look up V and K mags for TWA in Simbad
- Match both queries
- Create color-color diagram: NUV-V vs. V-K

cpsanchez committed:
Add files via upload

Updated tutorial:

- Addition of Hipparcos sample of older star for color-color comparison
- Exercise for the reader to recreate all the steps of the tutorial for older-stars sample + compare results
- Addition of color-coded scatter plot to display age information in color-color diagram

Issue: some targets that have been observed by Galex do not show up in GUVCat VO, Brian McLean is looking into this issue, for now we are just using those targets that can be accessed via coordinate query through GUVCat VO.

cpsanchez committed:
Add files via upload

Updates to tutorial:

- Changed old stars target
- Added query in VizieR catalog of 512 sources
- Changed structure of tutorial, now color-color is the goal of the tutorial
- Exercise for this tutorial is still TBD

cpsanchez committed:
Add files via upload

Updates to tutorial:

- Fixed long outputs to make it more readable
- Creation of exercise

cpsanchez committed:
Add requirements.txt

cpsanchez committed:
Add TWA_coordinates file

Complementary file for tutorial containing coordinates to almost all TWA members (to be used inside tutorial)

cpsanchez committed:
Add Solution's Notebook

cpsanchez committed:
Add files via upload

Updates to tutorial:

- Exercise without solution
- Brief explanation about TW Hya after exercise
- Correction of outliers

cpsanchez committed:
Add files via upload

Fixed typos inside Tutorial

dr-rodriguez commented:
I think this looks great. One thing I would perhaps change is that the mention that GALEX can be found at "the author's website" is very odd. GALEX is not a paper with an author and that's not even the mission website. I would rephrase that as `The GALEX catalog can be accessed via multiple places including this website, through MAST Casjobs, ...`

Also, we should probably list where we got the TWA stars from. I can't remember if I sent those to you or you grabbed them from a paper or from Simbad or elsewhere. Just for reproduceability in terms of the input data.

@ttdu you may want to have a look at this in general, too.

cpsanchez committed:
Fixed comments on tutorial

Changes made:

- Fixed wording about where GALEX can be found
- Specified where the coordinates for TWA come from -> SIMBAD

cpsanchez committed:
Typos' correction

cpsanchez commented:
Fixed it! Thank you for the comments @dr-rodriguez

ttdu commented:
Hey @cpsanchez! I have some feedback for you as well; I tried reviewNB, but I had multiple comments per cell, so I'll Slack you a notebook that has my suggestions edited in.

Otherwise, two bigger things to mention:
- The GALEX Database is different from GUVcat. The Database is a huge collection of all the sources GALEX has ever observed, whereas GUVcat is a cleaned version of the Database created by Bianchi et. al that excludes multiple data entries per source (and has additional science info). I think this distinction gets lost a little bit in your first two sections -- I would always say GUVcat when you mean GUVcat, rather than 'The GALEX Catalog'
- In the notebook I'm Slacking to you, I've added a table of contents. I'd recommend using that as a starting place to re-organize some of the content, like adding some "sub-chapters" to improve readability. You'll see what I mean in the "simplified example" table of contents. When possible, you should also make the names focused on the **action** in that section (i.e. 'Getting Magnitudes of Nearby M-Dwarfs' instead of 'Nearby M-Dwarfs')

The notebook is looking great so far, and I hope my comments help you polish things up!

cpsanchez committed:
Updated tutorial

Changes performed:
- Adopted new STScI template for tutorial
- Included table of contents + changed distribution
- Changed some terminology

cpsanchez committed:
Updated tutorial solution

Changes performed:
- Adopted new STScI template for tutorial

jrka commented on UV_excess_in_low_mass_stars/UV_excess_in_low_mass_stars.ipynb:
```diff
@@ -0,0 +1,948 @@
+{
```
<p>I'd recommend splitting these up into multiple cells.</p><p> </p><p>For the learning goals, can you add some more specificity? For example, for "Download data" I presume (after having read the intro) this will specifically be GALEX spectra and we'll be using astroquery to download it from the MAST archive? [Haha, nope! I was wrong. I came back to this comment after reading through the rest of the tutorial.] </p><p> </p><p>It seems the learning goals might be more like....</p><ul><li>Load a catalog of star coordinates from a text file</li><li>Load a catalog of star coordinates from Vizier</li><li>Search GUVcat and Simbad for photometry of target stars given coordinates</li><li>Create a color-color diagram of stellar populations and interpret the results</li></ul><p> </p><p>That might not be extensive, there is probably more here that I am missing because the tutorial is very comprehensive, but those are the ones that stood out to me upon first review. </p><p> </p><p>As the reader I was also hoping for a bit more at the end of the introduction. I'm left wondering, of all thse cool things we can do with GALEX(look for variable, detect binaries, detect chromospheres), are we actually going to do one in the tutorial? Which one?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

jrka replied:
<p>I am not at all familiar with pyvo! So this might also be a good thing to add to the introduction, give the reader the setup of 1) what they will be do (the science goals) and 2) the tools they will use for it.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

jrka replied:
<p>Recommend to change the title to say what we are going to do with GUVcat.</p><p> </p><p>Also need some information, either here or back up at the intro, about what GUVcat is and how we will use it.</p><p> </p><p>I appreciate the explanation of the naming conventon! </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

jrka replied:
<p>Why TWA 1? I realized through context this must be a star. So in the introduction, you could say that we plan to study TWA 1 and explain why you chose that one. Something interesting about it? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

jrka replied:
<p>Can you add a markdown cell above this code block that explains what you are doing? I can follow along that you are specifying the coordinates you are looking for and you want results in a radius of 10 arcseconds. But, for example, will the user always use that URL to search GALEX though VO? Can you link to documentation to where a new user would have found that URL in the first place?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

jrka replied:
<p>Oh, I had asked about TWA above, now I see it spelled out - be sure it's spelled out when the acronym is first used (above).</p><p> </p><p>I would make sure that either here, or ideally in the introduction above, a color-color diagram is defined and also its usefulness explained.</p><p> </p><blockquote>will be then to load this</blockquote><p> </p><p>I suggest remove "then".</p><p> </p><p>This section is really clearly explained with lots of detail! As the user I'm like, "Yes, I am following along!"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

jrka replied:
<p>In retrospect, should loading the catalog of TWA star locations come first, and then search VO? Because that would be the scientific workflow... I've got a big list of stars, I want to know which ones have GALEX data. You could still demonstrate how to search VO with just the one star at first, and then do the loop for the whole list of stars.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

jrka replied:
<p>Which bands do we have so far? FUV and NUV?</p><p> </p><p>I think we need to add what Simbad is, this is not raw data or observations we are querying, but published magnitudes. (I guess the same should be said about for GUVCat.)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

jrka replied:
<blockquote>below the name 'FLUX_K' it indicates it is a magnitude!</blockquote><p>below the name 'FLUX_K' the unit indicates it is a magnitude!</p><p> </p><p>Rather than "problematic" values, are they "missing" values? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

jrka replied:
<p>; instead of , before "we will"</p><p> </p><p>A bit more explanation will help here, since there are 20 lines of code coming up. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

jrka replied:
<p>Oops the output below says 41? Not sure if a typo or things changed! I found it helpful to not list specific numbers or outputs in my markdown cells in case they change in the future. So you can get around this by saying e.g. "We can check the length of the maK_TWA list to confirm that we have removed the missing values:"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

jrka replied:
<p>This is a good example of something that could be explained in the introduction. So we want to make a color-color diagram of two different populations: 1) the TW Hydra Asssociation, young stars whose coordinates are provided in a text file, and 2) a population of nearby M-dwarfs, old stars, the coordinates of which we'll get from a published catalog in Vizier. For both populations, we'll get the FUV, NUV magnitudes from GUVCat and the K, V magnitudes from Simbad.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

jrka replied:
<p>This is so interesting! This is also information that could be in the introduction. What am I looking for when I look at a color-color diagram?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='50'/>

jrka commented:
<p>I made a number of in-line comments in ReviewNB, but I think the main suggestion is to rewrite the intro (now that you have the rest of it done) to let the reader know what the science question is and what tools we'll use to answer the question.</p>

cpsanchez committed:
Corrected comments from Julia's review

Main changes to tutorial:

- Changed introduction 
- Included more explanations about the science goal and how we will go about it
- Changed structure of tutorial
- Some vocabulary changes

cpsanchez commented:
Thank you so much @jrka for all your comments, I believe I had taken all of them into consideration. Nevertheless I will come back to this tutorial in a couple of days to give it a second overall review and make sure all the recommendations have been implemented!

cpsanchez committed:
Updated old version of tutorial
